ci/tests/test_codec_cpu_audit.py

```
@@ -0,0 +1,195 @@
+    current["host_key"] = AUDIT.environment_key(current["environment"])
+    with pytest.raises(RuntimeError, match="host baseline is unknown"):
+        AUDIT.check_trend(baseline, current)
+
+
+def test_trend_gate_checks_callgrind_function_share() -> None:
+    baseline = AUDIT.load_trend()
+    current = json.loads(json.dumps(baseline))
+    functions = current["backends"]["helix"]["host"]["callgrind"]["functions"]
+    function = next(iter(functions))
+    functions[function] = functions[function] * 1051 // 1000
+    with pytest.raises(RuntimeError, match="callgrind-function/.+ regressed"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)